src/components/movie-list-page/movie-list-page.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect } from "react"
+import { useState } from "react";
+
+import { MoviesList } from "../movies-list/movies-list";
+import { HeaderSearch } from "../header-search/header-search";
+import { MovieDetails } from "../movie-details/movie-details";
+import { Portal } from "../portal/portal";
+import { EditMovieDetails } from "../edit-movie-details/edit-movie-details";
+import { ModalDialog } from "../modal-dialog/modal-dialog";
+import { nanoid } from "nanoid";
+import { DeleteMovie } from "../delete-movie/delete-movie";
+
+export interface MovieData {
+    id: string;
+    name: string;
+    imageUrl: string;
+    releaseYear: string;
+    duration: string;
+    relevantGenres: string[];
+    description: string;
+    rating: string;
+}
+
+const genres = ['Documentary', 'Comedy', 'Horror', 'Crime'];
+
+
+// const moviesArray: MovieData[] = [];
+
+// for (let i = 0; i < 1; ++i) {
+//     moviesArray.push({
+//         id: nanoid(),
+//         name: `Oppenheimer ${20 - i}`,
+//         imageUrl: 'https://avatars.mds.yandex.net/get-kinopoisk-image/4486454/c5292109-642c-4ab0-894a-cc304e1bcec4/600x900',
+//         releaseYear: 2025 + (-2 ^ i),
+//         duration: '3h 10min',
+//         relevantGenres: ['DOCUMENTARY', 'HORROR'],
+//         description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean turpis turpis, mollis ut cursus ac, ultricies sed odio. Duis eleifend elit quam, sed gravida odio pharetra in. Quisque quam est, condimentum nec turpis id, auctor vulputate dui. Vivamus tempus, arcu ultrices tempor congue, odio orci semper nulla, placerat facilisis odio urna id tellus.`,
+//         rating: '8.8',
+//     })
+// };
+
+
+
+export default function MovieListPage() {
+    const [movies, setMovies] = useState<MovieData[]>([]);
+    const [sortBy, setSortBy] = useState('');
+    const [activeGenres, setActiveGenres] = useState<string[]>([...genres]);
+    const [searchText, setSearchText] = useState('');
+    const [modalDialogParams, setModalDialogParams] = useState<any>(null);
+    const [movieDetails, setMovieDetails] = useState(null as any);
+
+
+
+    const fetchMoviesData = (signal: any) => {
+        fetch(
+            decodeURIComponent(`http://localhost:4000/movies?sortBy=${sortBy}&sortOrder=asc&search=${searchText}&searchBy=title&filter=${activeGenres.length === genres.length ? '' : activeGenres.join(',')}`),
+            { signal }
+        )
+            .then(response => {
+                return response.json()
+            })
+            .then(parsedResponse => {
+                setMovies(parsedResponse.data.map((movie: any): MovieData => {
```

Comment:
we can chain .then so you can transform data before `setMovies`

Reply:
done